actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -89,11 +92,20 @@ class ActionCable.Connection
 
     open: ->
       ActionCable.log("WebSocket onopen event")
-      @disconnected = false
-      @subscriptions.reload()
+      # This is in the case that the browser client wants
+      # to open the connection regardless of whether a
+      # protocol was negotiated successfully.
+      if @webSocket?.protocol not in ActionCable.PROTOCOLS
+        @close()
+      else
+        @disconnected = false
+        @subscriptions.reload()
 
-    close: ->
+    close: (event) ->
       ActionCable.log("WebSocket onclose event")
+      if @webSocket?.protocol not in ActionCable.PROTOCOLS
+        ActionCable.log("WebSocket is closing because there is no common protocol. Stopping ConnectionMonitor.")
+        @monitor.stop()        
```

Comment:
I guess if you could plug into Sprockets somehow, you could pull a new version of the classes (perhaps namespaced differently for each asset build) and then hot swap in the ActionCable classes and reconnect. It would be super cool, but yes a lot of trouble.

As a side note: Facebook was doing [something similar](http://developer.android.com/reference/dalvik/system/DexClassLoader.html) on Android for awhile, before Google made them stop by modifying the TOS for the Google Play Store.